src/posix/platform/udp.cpp

```diff
@@ -427,6 +435,86 @@ otError otPlatUdpSend(otUdpSocket *aUdpSocket, otMessage *aMessage, const otMess
     return error;
 }
 
+otError otPlatUdpJoinMulticastGroup(otUdpSocket *       aUdpSocket,
+                                    otNetifIdentifier   aNetifIdentifier,
+                                    const otIp6Address *aAddress)
+{
+    otError          error = OT_ERROR_NONE;
+    struct ipv6_mreq mreq;
+    int              fd;
+
+    VerifyOrExit(aUdpSocket->mHandle != nullptr, error = OT_ERROR_INVALID_ARGS);
+    fd = FdFromHandle(aUdpSocket->mHandle);
+
+    memcpy(&mreq.ipv6mr_multiaddr, aAddress->mFields.m8, sizeof(mreq.ipv6mr_multiaddr));
+
+    switch (aNetifIdentifier)
+    {
+    case OT_NETIF_UNSPECIFIED:
+        break;
+    case OT_NETIF_THREAD:
+        mreq.ipv6mr_interface = gNetifIndex;
+        break;
+    case OT_NETIF_BACKBONE:
+#if OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE
+        mreq.ipv6mr_interface = gBackboneNetifIndex;
+#else
+        ExitNow(error = OT_ERROR_NOT_IMPLEMENTED);
+#endif
+        break;
+    }
+
+    VerifyOrExit(setsockopt(fd, IPPROTO_IPV6, IPV6_JOIN_GROUP, &mreq, sizeof(mreq)) == 0 || error == EADDRINUSE,
```

Comment:
```suggestion
    VerifyOrExit(setsockopt(fd, IPPROTO_IPV6, IPV6_JOIN_GROUP, &mreq, sizeof(mreq)) == 0 || errno == EADDRINUSE,
```

Reply:
will `setsockopt()` here causes `errno == EADDRINUSE`?

Reply:
Good catch! It should be `errno`.